### ISSUE TITLE
[Effective Dart] Disallow single line if statements without braces

### ISSUE THREAD

**Issue by navaronbracke** [dart-lang/site-www#5580]
Today, the [curly_braces_in_control_flow_structures](https://dart.dev/tools/linter-rules/curly_braces_in_flow_control_structures) lint does not lint on the following statement

`if (arg == null) return defaultValue;`

Per the style guide, this is due to an exception to the rule:

```
There is one exception to this: an if statement with no else clause where the entire if statement and the then body all fit in one line. In that case, you may leave off the braces if you prefer.
```

However, in code review, I have had to enforce that _all_ control flow statements (regardless of line length for the full statement) have braces.
Thus, in my opinion, the statement "DO use curly braces for all flow control structures." is not correct, due to this exception.

Perhaps we can either change the `curly_braces_in_flow_control_structures` lint (and remove the exception to the rule in Effective Dart),
or add a new lint that lints specifically for the single line `ìf (condition) <body>;`

Context: https://github.com/dart-lang/linter/issues/4870#issuecomment-1954797714

**Comment by gnprice** [comment]
> Thus, in my opinion, the statement "DO use curly braces for all flow control structures." is not correct, due to this exception.

I agree the description is a bit misleading and would be good to clarify. As is, the summary line is quite definitive but in reality there's an important exception.

I think this would do it:
> DO use curly braces for all multi-line flow control structures.

If you want the other behavior, there's an existing lint for that: [always_put_control_body_on_new_line](https://dart.dev/tools/linter-rules/always_put_control_body_on_new_line). That's what the Flutter tree uses, for example. Combining that lint and this one gets you a requirement that all control-flow structures use curly braces.

In addition to clarifying what this lint does, it would probably be useful for this lint's description to mention the other one, to help out readers who want that other behavior.
